### ISSUE TITLE
Allow to append more statements

### ISSUE THREAD

**Issue by oscarotero** [shadowhand/latitude#58]
Hi. I need to create queries programatically, but seems that this library doesn't have methods to append more statements without remove the olders. For example:

```php
$query = $builder->select($basic_columns);

if ($more_columns) {
    $query = $query->columns($more_columns);
}
```

Adding more columns, removes the older columns. But this behaviour is not consistent. There's some functions that always add statements without override the previous. For example:

```php
$query = $builder->select($columns)
    ->join($table, $criteria)
    ->join($table, $criteria)
    ->orderBy($column)
    ->orderBy($column)
```

Other capabilities have methods to specifically append instead override, like `where`, `andWhere` and `orWhere`.

I suggest to have a consistent way to all statements. Some options:

* Override always and provide a `and` method to append. For example:
  ```php
  $query->orderBy($column)
        ->andOrderBy($other_column)
  ```
* Other option is append always. This makes the `andWhere` useless:
  ```php
  $query->column($basic_columns)
        ->column($more_columns)
        ->where($a_condition)
        ->where($other_condition)
  ```
* And finally, have a way to get the current value to manipulate it before override:
  ```php
  $columns = $query->getColumns();
  $columns[] = $new_column;
  $query->column($columns);
  ```

I don't have a strong preference for any option (and you may have a better idea), but would love to have a way to archieve this. What do you think?

**Comment by shadowhand** [comment]
Thanks for bringing this up. You're absolutely right about the inconsistency. I could certainly make all methods be append-only, but that would confusing if you want to (eg) replace columns.

I don't particularly like the idea of having getters for internal state, though it would allow Latitude to adopt a Visitor pattern to build up queries rather than be entirely internal, which would also for greater flexibility in customizing the SQL output. That is probably something I should target for the next major release.

In the mean time, I think the best course of action would be to make `columns()` be an append-only operation. The only oddity there is that the default column list is `*` and would need to be replaced by the first call to `columns()`.

**Comment by oscarotero** [comment]
Great! The most immediate need in my project is to be able to append columns and tables. That is, make `column()` and `from()` append-only.
Thank you!

**Comment by shadowhand** [comment]
Thinking about it more, I don't see how it would be possible to make `columns()` and `from()` be append-only without breaking backwards compatibility. For a minor release, we can only add new functionality, which would mean having new append methods or getters.

**Comment by oscarotero** [comment]
I understand.
My particular preference is create new append methods, like `andColumn()` and `andFrom()`. It's easier to use and cover the most needs because the common use is to build the query adding items, not removing or replacing them.

**Comment by shadowhand** [comment]
Fixed in version 3.1.0.

**Comment by oscarotero** [comment]
Thank you!
